live_data_subscriber/__init__.py

```diff
@@ -81,12 +60,15 @@ def ws_mapping(api_client_args, live_data_device_uuids, api_client_mapping,
             get_area_uuid_and_name_mapping_from_simulation_id(configuration_id, sim_api_domain_name)
         # It is a prerequisite to have unique area names for this to work.
         uuid_name_mapping = {v: k for k, v in name_uuid_mapping.items()}
-        if is_mqtt:
-            mqtt_mapping(api_client_args, live_data_device_uuids, api_client_mapping,
-                         uuid_name_mapping)
-        else:
-            ws_mapping(api_client_args, live_data_device_uuids, api_client_mapping,
-                       uuid_name_mapping)
+        for device_uuid in live_data_device_uuids:
+            device_name = uuid_name_mapping[device_uuid]
+            if device_name in allowed_devices_name_mapping:
+                key_name = allowed_devices_name_mapping[device_name] if is_mqtt else device_name
```

Comment:
you dont need the mqtt flag any more! That was the point actually

Reply:
Please give this commit another look: https://github.com/gridsingularity/d3a-api-client/commit/381250b20a941f95ba8c06591e3b13b6fd21d4a5
`is_mqtt`  is no more used inside `generate_api_client_args_mapping`